### DISCUSSION TITLE
Anyone using successfully with codespaces?

### DISCUSSION THREAD

**Discussion by sheldonhull** [twpayne/chezmoi#1631] [Q&A] [answered]
I've used the directions for bootstrap script.

Right now I seem to require running the init command with `https://me:token@github...` for it to work, as logs always show username required since my chezmoi repo is a private repo. I'm not certain how to ensure the bootstrapping process honors my git credentials so I can just have it automatically run chezmoi on new codespaces.

I also have set `Access and security` [settings](https://github.com/settings/codespaces) to all repositories to ensure full access in codespaces, but still doesn't solve the issue. 

Any ideas? 

I've really found chezmoi to help setup all my environments quickly, but having to manually run for each codespace is a bit frustrating. 

I'd love to open my chezmoi repo to the world, but leaned on the side of caution to avoid accidentally including any files with sensitive values unintentionally.

**Comment by felipecrs** [comment, marked as answer]
Codespaces is the one responsible for cloning the dotfiles repo here, not chezmoi. The purpose of the bootstrap script is to simply handover the rest of the job to chezmoi (like creating the first configuration file and running `apply` thereafter), after Codespaces initially cloned the repository.

The problem here is that Codespaces does not support private repositories as of now, [as they state in their documentation](https://docs.github.com/en/codespaces/customizing-your-codespace/personalizing-codespaces-for-your-account#dotfiles).

I would advise you to check if this feature request is already being tracked by them, and if yes, to upvote. Otherwise, to make the feature request itself.

**Reply by sheldonhull** [reply]
I've done a ton of reading on this in the past and totally missed the private not supported issue would impact using a custom dotfiles repo. I figured since I'm using "chezmoi" repo not dotfiles and have the executable that it would be able to do this.

Maybe the workaround is to use a name/token in the url in my own private synced settings file for dotfiles so it can reach it. Not ideal, but might work. (Note: I really wish GitHub had the concept of a token specific to a single repo). Otherwise, I'd need to make my chezmoi repo public, which I could do, just had hesitated on. 

thanks for the clarity!

**Reply by felipecrs** [reply]
Hm... from what I know you don't have the option to set a custom dotfiles repo url:

![image](https://user-images.githubusercontent.com/29582865/142255546-9defa9c3-fad5-4d61-904f-ef33c6204483.png)
